### ISSUE TITLE
urllib.request.urlopen('file:...').url incorrect value

### ISSUE THREAD

**Issue by barneygale** [python/cpython#127090]
# Bug report

### Bug description:

When a `file:` URL is given to `urllib.request.urlopen()`, it returns an [`addinfourl`](https://docs.python.org/3/library/urllib.request.html#urllib.response.addinfourl) object. This object's `url` attribute (and its deprecated `geturl()` method) usually return incorrect results. For example:

```python
>>> from urllib.request import urlopen
>>> urlopen('file:Doc/requirements.txt').url
'file://Doc/requirements.txt'   # expected: 'file:Doc/requirements.txt'
>>> urlopen('file:C:/requirements.txt').url
'file://C:/requirements.txt'  # expected: 'file:C:/requirements.txt' or 'file:///C:/requirements.txt'
```

The code always prepends `file://`, but this might be too many slashes or too few depending on the path's drive and root.

### CPython versions tested on:

CPython main branch

### Linked PRs
* gh-127091

**Comment by kanishka-coder0809** [comment]
Hi @barneygale I can work upon this issue ...Can you provide some more details about the issue.

**Comment by barneygale** [comment]
I'm sorry @kanishka-coder0809, I'm afraid I already opened a PR for this issue shortly before your comment: https://github.com/python/cpython/pull/127091

**Comment by barneygale** [comment]
Fixed in 3.14, but not worth backporting IMO. Closing...
